File: app.prueba/src/main/kotlin/com/example/app/prueba/entity/Lead.java
import jakarta.persistence.*;

@Entity
@Table(name = "leads")
public class Lead {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String email;
    private String phone;
    private String status;

    // Getters and Setters
}

File: app.prueba/src/main/kotlin/com/example/app/prueba/entity/Lead.kt
import jakarta.persistence.*

@Entity
@Table(name = "leads")
class Lead {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private val id: Long? = null

    @Column(nullable = false)
    private var name: String? = null

    @Column(nullable = false, unique = true)
    private var email: String? = null

    @Column(nullable = false)
    private var status: String? = null

    @OneToMany(mappedBy = "lead", cascade = [CascadeType.ALL], orphanRemoval = true)
    private val activities: List<Activity>? = null // Getters and setters
}